### ISSUE TITLE
Outbound settings for chain-proxying

### ISSUE THREAD

**Issue by TPS** [AdguardTeam/AdguardForAndroid#51]
As mentioned [here & preceding](35#issuecomment-141159277), outbound proxy settings are necessary for those of us who'd like to use AG in conjunction with another proxy, e.g., Orbot in that mode.

**Comment by ameshkov** [comment]
We should implement it as a low-level setting.

**Name:** pref.http.proxy.upstream
**Type:** String
**Example:** 127.0.0.1:8888

**Comment by ameshkov** [comment]
One more user request: add PAC support

**Comment by ameshkov** [comment]
@Revertron need a hand with this.

#### Step 1: Add "Proxy settings" to the advanced section:
<img width="1187" alt="2016-05-18_1758" src="https://cloud.githubusercontent.com/assets/5947035/15363967/4a608fca-1d23-11e6-86a5-bf13fe8139a7.png">

#### Step 2: Proxy settings dialog

It should look like the system dialog for Wi-Fi network properties. Fields are described below.

**Proxy type**
`None`
`HTTP`
`SOCKS4`
`SOCKS5`

**Proxy hostname** 
**Proxy port**
Visible for both HTTP and SOCKS.

**Username**
**Password**
Visible for HTTP and SOCKS5 (not for SOCKS4).
Validation rules: ASCII characters, can be empty.

**Cancel**
**Save**

**Comment by Revertron** [comment]
Done with dialog :)

**Comment by ameshkov** [comment]
Short instruction on how to use Adguard with Orbot (we should move add it to FAQ):
1. Install and run Orbot
2. Open Adguard and go to Settings -> Advanced
3. Set proxy settings:
SOCKS4
host=127.0.0.1
port=9095
4. Run Adguard

**Comment by ameshkov** [comment]
Done

**Comment by evgeniy-ADG** [comment]
Hi, could you please look into comments below, the customer cannot get Adguard work along with Adguard

<details>

I am having a difficulty in connecting to internet using both Orbot and
Adguard together.

Your previous e-mail didn't contain all the options necessary for using
Orbot with Adguard. In your email you said to use local proxy port which is
9095. But, how it is connected to Orbot? Also, Orbot's which setting is
connected to Adguard's local proxy port 9095. Meaning, not everyone is
using the Orbot's default settings. So, it is necessary to know settings
not just from Adguard's view but also Orbot's in order to correctly use
both apps together. So, if I want to change proxy port 9095 to something
else, then what should I have to change in Orbot's settings to make both
work.

My phone is not rooted, so I have to use transposing of apps in VPN mode to
use Orbot to be able to surf anonymously. But, the problem is if I activate
Adguard, then Orbot needs to be disconnected to use Internet.

Can I use Internet with or without VPN mode in either one or not? In
non-rooted phones, in order to use Orbot's anonymity in every apps, I have
to use transproxy support with VPN mode. Then only it becomes usable. Also,
I don't know outbound proxy network in Orbot app.

Now, another problem in Adguard is whether to use VPN mode or local proxy
mode in Adguard when using with Orbot. Because, the problem is only one app
can use VPN mode at a time.

If I change filter mode to "local http proxy in Adguard", then local proxy
port option becomes visible, and then again, another problem occurs is what
is that used for?

In Orbot's settings, under "Debug" menu, there are three options with exact
names in latest Orbot app. They are as follows:
Tor Socks
Tor transproxy Port
Tor DNS Port
So, I want to know that are these options important when using Adguard with
Orbot.

I have embedded few snapshots to get an overview of what I meant to say.
Especially the one which keeps saying "Orbot is starting ".

In short, there are many options we have to go through when using both.

Options to consider in both apps in brainstorm are as follow:-

1) VPN mode in either or both.
2) Orbot's transproxy feature should be enabled or not when using Adguard.
3) In Adguard, Using "VPN mode or local http proxy" in filter mode option.
4) Who is first using Internet, Orbot or Adguard? Meaning, is Adguard
getting Internet content from Orbot or is Orbit getting Internet content
from Adguard after filtering all ads?

**Comment by ameshkov** [comment]
@evgeniy-ADG I suppose there is a misunderstanding of some sort.

In order to work together, Adguard should use Orbot as an outbound proxy.

So, the chain looks like:
Adguard ---> Orbot ---> Internet

It does not depend on which filtering mode is used by Adguard.
I suppose user tries to change "Local proxy port" in AG settings instead of setting "Proxy settings".
